[UNIT] llama-index framework - release 4/3/25 (#22)
huggingface/agents-course: units/en/unit2/introduction.mdx

# Frameworks

> Now, we've mastered our basic understanding of agents, we can dive into the most popular frameworks that are used to build them!

Reviewer: ```suggestion
Now, we've mastered our fundamental understanding of agents, we can dive into the most popular frameworks that are used to build them!
```

In this second unit, **You will learn how to build agents and agentic workflows with various frameworks** including:

> **[LangGraph](https://github.com/langchain-ai/langgraph)**: a library for building stateful, multi-actor applications with LLMs, used to create agent and multi-agent workflows.

Reviewer 2: LangGraph should be at the bottom since it comes last, no?

> **[LangGraph](https://github.com/langchain-ai/langgraph)**: a library for building stateful, multi-actor applications with LLMs, used to create agent and multi-agent workflows.

Reviewer: Swap the order of LangGraph and LlamaIndex so it follows release order.

> **[smolagents](https://github.com/huggingface/smolagents)**: a barebones library for agents. Agents write python code to call tools and orchestrate other agents.
> 
> **[LangGraph](https://github.com/langchain-ai/langgraph)**: a library for building stateful, multi-actor applications with LLMs, used to create agent and multi-agent workflows.

Author: ```suggestion
**[LangGraph](https://github.com/langchain-ai/langgraph)**: a library for building stateful, multi-actor applications with LLMs, used to create agent and multi-agent workflows.

**[smolagents](https://github.com/huggingface/smolagents)**: a barebones library for agents. Agents write python code to call tools and orchestrate other agents.
```

**[LlamaIndex](https://github.com/run-llama/llama_index)**: a data framework for your LLM application. Building with LlamaIndex typically involves working with LlamaIndex core and a chosen set of integrations (or plugins).

Sounds more or less similar, right?

Don't worry. After this unit, you will have a solid understanding of each one of these frameworks and **how and when to use each framework**. You will also have **hands-on experience with building agents** using each framework.

> You will for example, create an agent, called Alfred, which can help you with agentic Retrieval Augmented Generation (RAG) tasks.

Reviewer: ```suggestion
You will create an agent called Alfred, which can help you with tasks like Agentic Retrieval Augmented Generation (Agentic RAG).
```
